xc7/utils/prjxray_create_edges.py

```diff
@@ -822,37 +815,34 @@ def main():
 
         edge_set = set()
 
-        for loc in progressbar.progressbar(grid.tile_locations()):
-            gridinfo = grid.gridinfo_at_loc(loc)
-            tile_name = grid.tilename_at_loc(loc)
+        c2 = conn.cursor()
+        for tile_pkey, tile_type_pkey, grid_x, grid_y in progressbar.progressbar(c.execute("SELECT pkey, tile_type_pkey, grid_x, grid_y FROM tile")):
+            loc = (grid_x, grid_y)
 
             # Not a synth node, check if in ROI.
-            if use_roi and not roi.tile_in_roi(loc):
+            if use_roi and not (roi_range[0] <= loc[0] and loc[0] <= roi_range[2] and roi_range[1] <= loc[1] and loc[1] <= roi_range[3]):
```

Comment:
Again, for the ROI, probably best to iterate over VPR tiles, but map back to phy coordinates and check against the phy ROI